torchgeo/models/weights.py

```diff
@@ -0,0 +1,154 @@
+"""Pretrained weights for models."""
+import os
+from typing import Any, Dict
+
+import torch
+import torch.nn as nn
+from torchvision.models._api import Weights, WeightsEnum
```

Comment:
I'm guessing this will require us to bump the minimum version of torchvision (and therefore torch)? I'm fine with that.